Index: tsuru-server-0.11.2/src/github.com/fsouza/go-dockerclient/container.go 
================================================================================
--- tsuru-server-0.11.2.orig/src/github.com/fsouza/go-dockerclient/container.go
+++ tsuru-server-0.11.2/src/github.com/fsouza/go-dockerclient/container.go
@@ -419,12 +419,19 @@
 	NetworkMode     string                 `json:"NetworkMode,omitempty" yaml:"NetworkMode,omitempty"`
 	IpcMode         string                 `json:"IpcMode,omitempty" yaml:"IpcMode,omitempty"`
 	PidMode         string                 `json:"PidMode,omitempty" yaml:"PidMode,omitempty"`
+	UTSMode         string                 `json:"UTSMode,omitempty" yaml:"UTSMode,omitempty"`
 	RestartPolicy   RestartPolicy          `json:"RestartPolicy,omitempty" yaml:"RestartPolicy,omitempty"`
 	Devices         []Device               `json:"Devices,omitempty" yaml:"Devices,omitempty"`
 	LogConfig       LogConfig              `json:"LogConfig,omitempty" yaml:"LogConfig,omitempty"`
 	ReadonlyRootfs  bool                   `json:"ReadonlyRootfs,omitempty" yaml:"ReadonlyRootfs,omitempty"`
 	SecurityOpt     []string               `json:"SecurityOpt,omitempty" yaml:"SecurityOpt,omitempty"`
 	CgroupParent    string                 `json:"CgroupParent,omitempty" yaml:"CgroupParent,omitempty"`
+	Memory          int64                  `json:"Memory,omitempty" yaml:"Memory,omitempty"`
+	MemorySwap      int64                  `json:"MemorySwap,omitempty" yaml:"MemorySwap,omitempty"`
+	CPUShares       int64                  `json:"CpuShares,omitempty" yaml:"CpuShares,omitempty"`
+	CPUSet          string                 `json:"Cpuset,omitempty" yaml:"Cpuset,omitempty"`
+	CPUQuota        int64                  `json:"CpuQuota,omitempty" yaml:"CpuQuota,omitempty"`
+	CPUPeriod       int64                  `json:"CpuPeriod,omitempty" yaml:"CpuPeriod,omitempty"`
 }
 
 // StartContainer starts a container, returning an error in case of failure.
