Index: tsuru-server-0.11.3/src/github.com/tsuru/tsuru/api/server.go
=============================================================================================
--- tsuru-server-0.11.3.orig/src/github.com/tsuru/tsuru/api/server.go	2015-07-31 11:34:33.825178000 -0300
+++ tsuru-server-0.11.3/src/github.com/tsuru/tsuru/api/server.go	2015-07-31 11:34:57.909217883 -0300
@@ -26,7 +26,7 @@
 	"gopkg.in/tylerb/graceful.v1"
 )
 
-const Version = "0.11.2"
+const Version = "0.11.3"
 
 func getProvisioner() (string, error) {
 	provisioner, err := config.GetString("provisioner")
