Author: Luk Claes <luk@debian.org>
Description: Fixes FTBFS with --no-add-needed (Closes: #554059).

--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@ TESTS = fsck-tests.sh
 INSTALL = install
 prefix ?= /usr/local
 bindir = $(prefix)/bin
-lib_LIBS = -luuid -lblkid -lm -lz -llzo2 -L.
+lib_LIBS = -luuid -lblkid -lm -lz -llzo2 -lcom_err -L.
 libdir ?= $(prefix)/lib
 incdir = $(prefix)/include/btrfs
 LIBS = $(lib_LIBS) $(libs_static)
