Description: Use correct ELF header for armhf binaries.
Author: Adam Conrad <adconrad@ubuntu.com>
Last-Update: 2013-07-08

Index: golang-1.1/src/cmd/ld/elf.c
===================================================================
--- golang-1.1.orig/src/cmd/ld/elf.c	2013-07-08 08:25:34.246800925 -0600
+++ golang-1.1/src/cmd/ld/elf.c	2013-07-08 08:25:56.254801274 -0600
@@ -57,7 +57,11 @@
 	case '5':
 		// we only use EABI on linux/arm
 		if(HEADTYPE == Hlinux)
-			hdr.flags = 0x5000002; // has entry point, Version5 EABI
+#ifdef __ARM_PCS_VFP
+			hdr.flags = 0x5000402; // has entry point, Version5 EABI, hard-float ABI
+#else
+			hdr.flags = 0x5000202; // has entry point, Version5 EABI, soft-float ABI
+#endif
 		// fallthrough
 	default:
 		hdr.phoff = ELF32HDRSIZE;	/* Must be be ELF32HDRSIZE: first PHdr must follow ELF header */
