Description: Source file to fix http://bugs.debian.org/716853
Author: Michael Stapelberg <stapelberg@debian.org>
Last-Update: 2013-12-31
Forwarded: https://codereview.appspot.com/46780043

Index: golang/src/pkg/debug/elf/testdata/hello.c
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ golang/src/pkg/debug/elf/testdata/hello.c	2013-12-31 18:25:09.268498582 +0100
@@ -0,0 +1,7 @@
+#include <stdio.h>
+
+void
+main(int argc, char *argv[])
+{
+	printf("hello, world\n");
+}
