Index: lxc-docker-0.11.1/daemon/execdriver/lxc/lxc_template.go
===================================================================
--- a/daemon/execdriver/lxc/lxc_template.go	2013-12-16 15:03:32.000000000 -0200
+++ b/daemon/execdriver/lxc/lxc_template.go	2014-01-17 14:36:10.257688903 -0200
@@ -98,9 +98,7 @@
 lxc.cgroup.memory.memsw.limit_in_bytes = {{$memSwap}}
 {{end}}
 {{end}}
-{{if .Resources.CpuShares}}
-lxc.cgroup.cpu.shares = {{.Resources.CpuShares}}
-{{end}}
+lxc.cgroup.cpu.shares = 460
 {{if .Resources.Cpuset}}
 lxc.cgroup.cpuset.cpus = {{.Resources.Cpuset}}
 {{end}}
