Index: lxc-docker-0.7.1/src/github.com/dotcloud/docker/lxc_template.go
===================================================================
--- lxc-docker-0.7.1.orig/src/github.com/dotcloud/docker/lxc_template.go	2013-12-16 15:03:32.000000000 -0200
+++ lxc-docker-0.7.1/src/github.com/dotcloud/docker/lxc_template.go	2014-01-17 14:36:10.257688903 -0200
@@ -132,9 +132,7 @@
 lxc.cgroup.memory.memsw.limit_in_bytes = {{$memSwap}}
 {{end}}
 {{end}}
-{{if .Config.CpuShares}}
-lxc.cgroup.cpu.shares = {{.Config.CpuShares}}
-{{end}}
+lxc.cgroup.cpu.shares = 460
 
 {{if (getHostConfig .).LxcConf}}
 {{range $pair := (getHostConfig .).LxcConf}}
