Index: node-hipache-deb/node_modules/hipache/lib/cache.js
===================================================================
--- node-hipache-deb.orig/node_modules/hipache/lib/cache.js	2013-12-06 16:19:41.298890073 -0200
+++ node-hipache-deb/node_modules/hipache/lib/cache.js	2013-12-09 11:08:56.578531495 -0200
@@ -31,9 +31,24 @@
             this.config.redisPort,
             this.config.redisHost
             );
+
+    if (this.config.redisMasterPort && this.config.redisMasterHost) {
+        this.redisMasterClient = redis.createClient(
+                this.config.redisMasterPort,
+                this.config.redisMasterHost
+                );
+    }
+
     this.redisClient.on('error', function (err) {
         this.log('RedisError ' + err);
     }.bind(this));
+
+    if (this.redisMasterClient) {
+        this.redisMasterClient.on('error', function (err) {
+            this.log('RedisMasterError ' + err);
+        }.bind(this));
+    }
+
     // Monitor active checker every 30 seconds
     this.monitorActiveChecker();
     setInterval(this.monitorActiveChecker.bind(this), 30 * 1000);
@@ -77,7 +92,11 @@
  */
 Cache.prototype.markDeadBackend = function (backendInfo) {
     var frontendKey = 'dead:' + backendInfo.frontend;
-    var multi = this.redisClient.multi();
+    if (this.redisMasterClient) {
+        var multi = this.redisMasterClient.multi();
+    } else {
+        var multi = this.redisClient.multi();
+    }
     // Update the Redis only if passive checks are enabled
     if (this.passiveCheck === true) {
         multi.sadd(frontendKey, backendInfo.backendId);
